nemo/collections/asr/parts/submodules/fast_rnnt_greedy_decoding.py

```diff
@@ -0,0 +1,440 @@
+                while_loop_args = np.array(
+                    [
+                        while_loop_conditional_handle.getPtr(),
+                        not_blank_ptr.ctypes.data,
+                        symbols_added_ptr.ctypes.data,
+                        max_symbols_ptr.ctypes.data,
+                    ],
+                    dtype=np.uint64,
+                )
+                with with_conditional_node(while_loop_kernel, while_loop_args, while_loop_conditional_handle):
+                    g, hidden_prime = self.caller._pred_step(
+                        self.last_label.unsqueeze(1), hidden, batch_size=self.batch_size
+                    )
+                    logp = self.caller._joint_step(self.f, g, log_normalize=None)[:, 0, 0, :]
+
+                    v, k = logp.max(1)
+                    torch.where(self.blank_mask, self.zero_t, v, out=v)
+                    torch.where(self.blank_mask, self.blank_index_t, k, out=k)
+                    # Commented out code unnecessarily causes D2H copy, which is synchronous. See pytorch issue #105641
+                    # self.scores[self.seq_idx_t, :] = v
+                    # self.labels[self.seq_idx_t, :] = k
+                    self.scores.index_copy_(0, self.seq_idx_t, v.unsqueeze(0))
+                    self.labels.index_copy_(0, self.seq_idx_t, k.unsqueeze(0))
+
+                    self.blank_mask.logical_or_(k == self.caller._blank_index)
+
+                    not_blank_mask = ~self.blank_mask
+
+                    self.caller.decoder.batch_copy_states_mask(hidden, hidden_prime, not_blank_mask)
+                    torch.where(self.blank_mask, self.last_label, k, out=self.last_label)
+
+                    torch.any(not_blank_mask, 0, out=self.not_all_blank_t)
+                    self.symbols_added_t += 1
+                    self.seq_idx_t += 1
+
+                self.time_idx_t += 1
+                self.seq_idx_t += self.max_symbols_t - self.symbols_added_t
+
+            self.scores_cpu.copy_(
+                self.scores.transpose(0, 1).contiguous().reshape((self.batch_size, self.max_time, self.max_symbols)),
+                non_blocking=True,
+            )
+            self.labels_cpu.copy_(
+                self.labels.transpose(0, 1).contiguous().reshape((self.batch_size, self.max_time, self.max_symbols)),
+                non_blocking=True,
+            )
+
+            self.last_label.fill_(self.caller._SOS)
+            self.time_idx_t.fill_(0)
+
+    def __call__(
+        self,
+        x: torch.Tensor,
+        out_len: torch.Tensor,
+        device: torch.device,
+        partial_hypotheses: Optional[List[rnnt_utils.Hypothesis]] = None,
+    ):
+
+        # Need to copy x and out_len into "staging buffers", or do a graph update.
+
+        if partial_hypotheses is not None:
+            raise NotImplementedError("`partial_hypotheses` support is not supported")
+
+        assert not self.caller.preserve_alignments
```

Comment:
Maybe `raise NotImplementedError` is better?

Reply:
I agree. Thank you for pointing it out.